okhttp/src/main/java/okhttp3/CipherSuite.java

```diff
@@ -29,356 +32,421 @@
  * <p>See also <a href="https://android.googlesource.com/platform/external/conscrypt/+/master/src/main/java/org/conscrypt/NativeCrypto.java">NativeCrypto.java</a>
  * from conscrypt, which lists the cipher suites supported by Android.
  */
-public enum CipherSuite {
+public final class CipherSuite {
+  // for interning - must be at top of file
+  private static final ConcurrentMap<String, CipherSuite> INSTANCES = new ConcurrentHashMap<>();
```

Comment:
(I might later look at replacing this with a copy-on-write LinkedHashMap; ConcurrentMap might be overkill for this seldom-edited case. As is this is fine.)